File: apps/src/chat/backend/packets/websocket/join_message_packet.py
from ..packet import Packet

class JoinMessagePacket(Packet):

    def __init__(self, sender: str):
        self._sender = sender

    def to_data(self):
        return {
            "sender": self._sender,
            "type": "join_message"
        }

    @staticmethod
    def from_data(data):
        return JoinMessagePacket(
            data["sender"]
        )

File: apps/src/chat/frontend/js/packets/websocket/JoinMessagePacket.js
export default class JoinMessagePacket {

    constructor(sender=""){
        this.sender = sender;
    }

    toData(){
        return {
            type: "join_message",
            sender: this.sender
        }
    }

    static fromData(data){
        return new JoinMessagePacket(data["sender"]);
    }

    getUsername(){
        return this.sender;
    }
}